gsw_xarray/tests/test_gsw_xarray.py

```diff
@@ -1,10 +1,69 @@
 from gsw_xarray import __version__
 import gsw_xarray as gsw
+import xarray as xr
+import pytest
+
 
 def test_version():
     assert __version__ == "0.1.0"
 
+
 def test_func_standard(ds):
     """Basic test"""
     sigma0 = gsw.sigma0(SA=ds.SA, CT=ds.CT)
-    assert sigma0.attrs['standard_name'] == 'sea_water_sigma_t'
+    assert sigma0.attrs["standard_name"] == "sea_water_sigma_t"
+    assert sigma0.name == "sigma0"
+    assert sigma0.attrs["units"] == "kg m^-3"
+
+
+def test_func_return_tuple(ds):
+    (CT_SA, CT_pt) = gsw.CT_first_derivatives(ds.SA, 1)
+    assert CT_SA.name == "CT_SA"
+    assert CT_SA.attrs["units"] == "K (g/kg)^-1"
+
+
+@pytest.mark.parametrize("gsdh", [0, 1, None])
+@pytest.mark.parametrize("ssg", [0, 1, None])
+@pytest.mark.parametrize(
+    "use_kw",
+    [
+        [
+            False,
+        ]
+        * (4 - i)
+        + [
+            True,
+        ]
+        * i
+        for i in range(5)
+    ],
+)
+def test_check_func(gsdh, ssg, use_kw):
```

Comment:
with the parametrization of pytest, this function tests all possibilities of call to gsw.z_from_p (using kwargs, args or no argument for the optional arguments). I wanted to test the check_function for z_from_p properly as I was not sure it would work in all case.